### DISCUSSION TITLE
ACHTUNG: Parallele Masseverbindungen

### DISCUSSION THREAD

**Discussion by SW-Niko** [hoylabs/OpenDTU-OnBattery#1374] [Show and tell]
Noch ein Hinweis für alle, die in der Elektrotechnik nicht so tief drinstecken.

Vermeidet bei der Verdrahtung zum ESP32 parallele Verbindungen über Steuerleitungen.
Falls ihr in dem folgenden Schaltbild die Gefahr nicht erkennt, dann solltet ihr vielleicht das verlinkte PDF anschauen.

**UPDATE: 01.12.24:** Neues PDF, Neues Schaltbild, Entstanden aus den unten geführten Diskussionen.

[ParalleleVerbindungen.pdf](https://github.com/user-attachments/files/17969973/ParalleleVerbindungen.pdf)

![grafik](https://github.com/user-attachments/assets/10f7cdb8-a48f-4ca3-8949-b06529a88875)

**Comment by spcqike** [comment]
Ja sieht nachvollziehbar aus. 

Ein paar real world Beispiele wären vielleicht gut. Sowohl wo Probleme herkommen können (hauptsächlich wohl, wenn das BMS abschaltet, was wohl in 99% der Fälle mit N-FETs passiert, aber auch wenn man umbaut und einfach so zuerst die Batterie Masse abklemmt) als auch, was dann passieren kann. Hatten wir da nicht letztens irgendwo ein Beispielbild? :)

**Reply by Snusme83** [reply]
@SW-Niko: Finde es super, dass du dich diesem wichtigen Thema mit im schlimmsten Fall verheerenden Folgen (Brand) annimmst!

Problem ist wahrscheinlich, nicht jeder kann was mit Begriffen wie einer seriellen Schnitstelle, Steuerleitungen anfangen. Genauso wie habe ich jetzt ein BMS mit N-FET?🤔

Ich glaube auch dass nur wenigen die verheerenden Folgen klar sind!  Wie spcqike schon erwähnte, hätte das besagte Beispielbild mit den abisolierten Adern eine präventive Wirkung!

Mit Beispielkonstellationen, welche hier gängig sind, könnte man bestimmt mehrere oDTUoBler erreichen! Dass es keine abschließende Auflistung von Beispielen geben wird, ist klar, aber mit den gängigsten Beispielen dürfte es klarer werden.

Ich für mich, schiebe es auch vor mir hin und hoffe dass das BMS meines/ meiner Pylontech keinen Sch... macht. Zudem betreibe ich mein Breakoutboard (auf dem der ESP32-S3 steckt) noch via Stepdown- Wandler über die Busbar!
Ist auch nicht richtig, aber ich habe drei Laderegler Victron SmartSolar 100/20 und mir ist noch nicht klar, wie ich den Adum1201 korrekt dazwischen installiere!🤷🏻‍♂️

**Comment by schlimmchen** [comment]
Sehr guter Beitrag, vielen Dank @SW-Niko. Ich begreife erst jetzt, warum @hoschiking so eindringlich davor gewarnt hat, dass bei eingreifendem BMS Strom über andere Massepfade fließen kann. Mir erschien das plausibel, aber erklären hätte ich es nicht können.

Darf man das klauen, übersetzen lassen, und in die Doku einbauen?

Eine kritische Rückfrage hätte ich allerdings noch: Welche Relevanz hat das in der Praxis, wenn das GND der VE.Direct Schnittstellen galvanisch getrennt ist? Welche relevante Masseschleife kann man denn dann bauen unter der Annahme, dass der ESP32 "wenigstens" über das BMS mit der Batteriemasse kurzgeschlossen ist? Selbst wenn man dann noch seinen ESP32 über einen DC-DC-Wandler aus der Batterie speist, gibt es kein Problem. Das kann ja erst entstehen, wenn man die Masse des ESP32 mit der Masse *verschiedener* Geräte verbindet, die dann auch potentiell jeweils mit dem positiven Batteriepol verbunden sind und Leistung in die Batterie schieben oder entnehmen.

![image](https://github.com/user-attachments/assets/9e327d50-e7c4-4248-b96b-00ce0627c248)

"little protection"... Aha?!

Also ich würde es jedenfalls als weiteres Argument für den Einsatz von ADUM1201 nehmen in der Doku.

**Reply by SW-Niko** [reply]
>Darf man das klauen, übersetzen lassen, und in die Doku einbauen?

Aber klar. 👍 
Ich bin auch bereit daran mitzuarbeiten. Kannst auch gerne die Open-Office Dateien haben.

>GND der VE.Direct Schnittstellen galvanisch getrennt ist.

Nicht nur GND, die gesamte Schnittstelle muss entweder galvanisch getrennt sein oder die Batteriespannung vertragen.

>Selbst wenn man dann noch seinen ESP32 über einen DC-DC-Wandler aus der Batterie speist, gibt es kein Problem. Das kann ja erst entstehen, wenn man die Masse des ESP32 mit der Masse verschiedener Geräte verbindet, die dann auch potentiell jeweils mit dem positiven Batteriepol verbunden sind und Leistung in die Batterie schieben oder entnehmen.

Nicht ganz so gefährlich, aber auch hier kann es zu Problemen kommen. Ich mache eine Zeichnung dann wird es klarer.

>Also ich würde es jedenfalls als weiteres Argument für den Einsatz von ADUM1201 nehmen in der Doku.

Prima. Das Thema sehe ich brandgefährlich. Man baut alles zusammen und es funktioniert wunderbar und wenn das BMS abschaltet brennt die Hütte ab.

**Reply by Snusme83** [reply]
@schlimmchen:
Hätte nicht gedacht, dass dir die Tragweite so nicht bewusst ist, aufgrund der Kompetenz, die du hier rüber bringst!

Was ich etwas schade finde hier: Das Wiki, etc. findet mit Masse im Deutschsprachigen Raum statt, aber dennoch wird es darauf hinausgelegt, mehr englischsprachig darzustellen.

Das "Englische" ist mir nicht ganz fremd, schaffe es auch unverständliches mir adäquat übersetzen zu lassen, aber wieso ist ned alles in deutsch und wenn man schon mehrsprachig unterwegs sein will, dies dann entweder ins Englische übersetzt oder deutsch verfestigt und die Benutzer es sich dann dementsprechend übersetzen lassen, falls es nicht im "Englischen" vorliegt!?!

**Reply by SW-Niko** [reply]
Hallo @schlimmchen,

ich habe jetzt in mein PDF noch ein Beispiel eingefügt.

[ParalleleMasse2.pdf](https://github.com/user-attachments/files/17669362/ParalleleMasse2.pdf)

**Reply by Sorbat** [reply]
Wie verhält sich das denn mit dem ISO Shield vom Fusion? Da sind doch schon Adums drauf

**Reply by schlimmchen** [reply]
korrekt.

**Reply by SW-Niko** [reply]
Hallo @schlimmchen ,
ich muss mein PDF nochmal überarbeiten.
Bei dem Beispiel mit dem DCDC Wandler und der BMS - ESP32 Verbindung betrachte ich nur die Energie, die für die ESP32-PCBA benötigt wird. Und das ist falsch. Man muss auch die MPPT und Inverter Energie betrachten und dann ist die Gefahr genauso hoch wie bei der MPPT - ESP32 Verbindung.
Vielen Dank an @hoschiking, deine Anmerkungen haben mich dazu angetrieben nochmal über die Problematik nachzudenken.

Jedenfalls nutze ich erst mal die neugewonnenen Erkenntnisse und gehe das Sicherheitskonzept meiner Anlage nochmal durch.

**Comment by hoschiking** [comment]
Das obige ist soweit alles korrekt, aber was nicht vergessen werden darf, dass in diesem speziellen Fall die Gefahr hauptsächlich entsteht, wenn ein BMS, wie das von JK verwendet wird und die UART nicht galvanisch getrennt wird. Sobald das BMS abgeschaltet, fließt der volle Batteriestrom über die Datenschnittstelle, wenn diese mit dem ESP verbunden ist. Kommen dann einer oder mehrere MPPT zum Einsatz,wird das ganze erst Recht kompliziert und die Gefahr von GND Loops ist nochmals größer. Das kann sehr schnell sehr unübersichtlich werden, daher würde ich grundsätzlich immer einen ADUM für die Schnittstellen verwenden, egal um was es sich handelt, insbesondere jedoch gilt es für das BMS.

**Reply by gitisgreat2023** [reply]
@swingstate @alexseuf sind eure PCBs bzgl dieses Thema sicher?

**Reply by Manos1966** [reply]
Die @swingstate PCBs haben ADUM fuer jeden VE.Direkt Eingang.

**Reply by gitisgreat2023** [reply]
@Manos1966  und für den RJ45?

**Reply by Manos1966** [reply]
> und für den RJ45?

Es ist mir noch nicht verstaendlich wann eine Masse Verbindung am RJ-45 benutzt wird 🤔 
Ich nutze den RJ45 fuer die CANbus Verbindung, sowie fuer die RS485 Verbindung (nicht gleichzeitig).
In beiden Faellen habe ich KEINE Masse angeschlossen!

**Reply by gitisgreat2023** [reply]
Ich stecke lediglich einen LAN Kabel rein... also alle Adern werden genutzt...

**Reply by Manos1966** [reply]
> Ich stecke lediglich einen LAN Kabel rein... also alle Adern werden genutzt...

Guter Punkt!
@swingstate @alexseuf ist RG45 PIN6 auf der Platine angeschlossen?

![Pylon RJ45](https://github.com/user-attachments/assets/3d1086ca-c7bf-4389-9a6d-da953987810e)

@gitisgreat2023 Ich nutze zwei einfachere Loesungen 😉 
![image](https://github.com/user-attachments/assets/e77a7ebc-1d3c-467e-a705-be22f700ce5a)

![image](https://github.com/user-attachments/assets/781e30d6-5490-4571-8444-cc7f8c20b419)

**Reply by SW-Niko** [reply]
Hmmm... ich merke langsam, das ich noch deutlicher werden muss. 🤔 
Ist aber auch meine Schuld wenn ich immer nur von parallelen Masse-Verbindungen spreche. Das ist falsch. 
Man muss eigentlich ganz allgemein von parallelen Verbindungen zum Batterie Potential sprechen.

Die gesamte Schnittstelle muss entweder galvanisch getrennt sein oder die Batteriespannung vertragen.

- Nur die Masse weglassen hilft nicht
- Sicherung in der Masse der Steuerleitung einbauen hilft nicht
- CAN und RS485 hilft nicht, wenn sie nicht galvanisch getrennt sind oder die Batteriespannung vertragen.

Ja ... das ist vielleicht nicht zu erkennen, auf den ersten Blick. Eine "normale RS485" arbeitet mit symmetrischer Übertragung und verträgt Spannungen gegenüber Masse bis -9V und +12V. Ich habe mir die CAN / RS485 Treiber ICs noch nicht im Detail angeschaut, aber ich bezweifle das sie mit 48V klar kommen.

Aber eine einzige nicht galvanisch getrennte Verbindung ist nach wie vor kein Problem (Alle weiteren Verbindungen müssen galvanisch getrennt sein).

**Reply by gitisgreat2023** [reply]
@SW-Niko "Aber eine nicht galvanisch getrennte Verbindung ist nach wie vor kein Problem."

Entweder soll der "nicht" raus oder der "k" bei kein oder?!

**Reply by SW-Niko** [reply]
Danke für den Hinweis. Ich habe es oben korrigiert.

**Reply by gitisgreat2023** [reply]
Bzgl die eine nicht galvanisch getrennte, das wäre dann typisch die Stromstärkste Verbindung, richtig? Batterie typischerweise. Oder gar keine wenn's nur Kommunikationssachen sind.

**Reply by SW-Niko** [reply]
Eigentlich egal. Vermutlich ist das trennen der Kommunikation das Einfachere.
Ich habe bei meiner Anlage die 3 VE.Direct Schnittstellen von der ESP32-PCB zu meinen 2 x MPPT und 1 x SmartShunt galvanisch getrennt.
Und die Stromversorgung für die ESP32-PCB mach ich über einen nicht galvanisch getrennten DCDC Wandler aus der Batterie.

Damit habe ich eine einzige Verbindung nicht galvanisch getrennt und alle anderen sind galvanisch getrennt. Passt.

**Reply by swingstate** [reply]
Sehr interessant. Ich arbeite gerade an einer neuen Board-Revision. Alle 3 Ve.Direct sind galvanisch getrennt. Der RJ45, der sowohl CAN als auch RS485 anbinden kann, bisher nicht. Sofern ich das Thema bisher richtig verstanden habe @SW-Niko müsste ich sowohl für CAN als auch für den RS485 IC jeweils einen ADUM zwischen RJ45  und dem jeweiligen ICs einbauen?

**Reply by SW-Niko** [reply]
Nein, nicht unbedingt. Eine von den Verbindungen muss nicht galvanisch getrennt sein.
Wenn du 3xVE:Direct + 1xPower + 1xCAN an der ESP32_PCB hast, dann kannst du davon eine direkt (ohne galvanische Trennung) verbinden.
Also 3xVE:Direct (ADUM) + 1xPower (USB Stecker-Netzteil) + 1xCAN (direkt, nicht galvanisch getrennt)

**Comment by hoschiking** [comment]
Ich versuche mal etwas Klarheit zu schaffen, denn ich habe den Eindruck, es wird kompliziter gemacht, als es ist.
Wenn wir von einer galvanischer Trennung einer Schnittstelle sprechen, zB einer UART, dann komplett und nicht nur GND.
Alles andere ist noch schlimmer und ihr zerschießt euch die Platinen, mit ganz wenig Pech.
Alle Geräte in einem System müssen bei massebezogenen Schnittstellen auf ein Potenziel gebracht werden, sonst entstehen Potenzialunterschiede. So eine UART ist bestenfalls 5V tolerant, verbindet man die GND nicht miteinader, dann kann zwar kein hoher Strom im Fall der Fälle fließen, jedoch können Potentialunterschiede unter den Geräten entstehen, alles >5V führt zum defekt der UART zumindest am ESP. Also wenn galvanisch trennen, dann komplett, ist ja mit einem ADUM ohnehin getan, oder ALLE Geräte und Schnittstellen auf ein Potential bringen, dann hat man "nur noch" das Problem mit der Abschaltung des BMS und der Masseschleife durch die Schnittstellen. Ausnahme bilden Schnittstellen die nicht Massebezogen sind, wie RS485, ein Potenzialunterschied ist in der Regel kein Problem, da Differenzsignal und nicht massebezogen -> die Schnittstelle kann nicht zerstört werden. Jedoch können Potenzialunterschiede dennoch zu Schäden führen, da hilft nur das ganze Gerät wenn möglich, potenzialfrei zu verschalten. Z.b. den ESP32 mit einem USB-Steckernetzteil, die meisten sind potenzialfrei, jedoch nicht immer. 
Es ist wie mit der PE-Schiene im Haus. Alles auf ein Potenzial zu bringen ist eine der wichtigsten Schutzmaßnahmen.
In Summe ist es sehr einfach -> Alle UART mittels ADUM galvanisch trennen (BMS, MPPT usw) fertig.
So viel Aufwand ist es auch nicht.

**Reply by swingstate** [reply]
Demnach ist für CAN ein ADUM notwendig, für RS485 nicht? In meinem Fall der RS485 Transceiver schon eine Verbindung zu GND.  Lt. Datenblatt ist der 1201 auch für RS485 vorgsehen.

**Reply by hoschiking** [reply]
Es kommt auf die verwendete Schnittstellenplatine und die Spannunsfestigkeit des IC an, das kann man pauschal nicht sagen.
Der übliche  RS485 von Amazon muß geräteseitig mit GND verbunden werden, auf der RS485 Seite scheint keine der beiden Datenleitungen mit GND verbunden zu sein, der scheint also galvanisch zu trennen. Mangelsdatenblatt, da Billigplatine für 3€ ist es nur eine Vermutung. Es kommt aber auf die Spannungsfestigkeit des Übertragers und die lokalen Gegebenheiten an. Wenn die verbundenen ordentlich auf der primären Seite (netzseitig) geerdet sind, sollte es keine großen Potenzialunterschiede geben.
Siehe Screenshot anbei. Kannst ja versuchen die chinesische ÜBersetzung zu interpretieren ;-)
Auch auf RS485 Seite gibt es einen Pin zur Erdung. Den kann man anschließen, muß man aber nicht. Sobald den PIN verbindest, dann unbedingt auf beiden Seiten der Zweidrahtstrecke, dann liegen beide auf gleichem Potenzial, dann ist es sicher vor großen Potenzialunterschieden, die galvanische Trennung ist dahin. Lässt man es weg, bist du galvanisch getrennt mit dem Restrisiko, dass der Potenzialunterschied zwischen den Geräten die Spannunsfestikgeit der Schaltung übersteigt. Im industriellen Umfeld mit goßen Strecken ist die Erdung also Pflicht, bei dir im Keller bei 1m Meter Entfernung eher zu vernachlässigen. die 50V Differenz sollte es verkraften, wenn das BMS trennen sollte. TVS Dioden und Fuses sind immerhin verbaut. Ohne Gewähr ;-)
![RS485](https://github.com/user-attachments/assets/249c3444-0cc8-4218-905c-79ab50655480)

**Comment by Manos1966** [comment]
> Die gesamte Schnittstelle muss entweder galvanisch getrennt sein oder die Batteriespannung vertragen.
Nur die Masse weglassen hilft nicht
Sicherung in der Masse der Steuerleitung einbauen hilft nicht
CAN und RS485 hilft nicht, wenn sie nicht galvanisch getrennt sind oder die Batteriespannung vertragen.

Also jetzt verstehe ich "Bahnhof" hier 🙈 

CAN und RS485 haben keine Masse, sondern High und Low.
Sie brauchen keine Masse Verbindung damit das Signal uebertragen wird. 

Ich muss die Plaene der CANbus und RS485 Platinen angucken...
Der Bedarf einer galvanischen Trennung  beim Victron MPPT ist mir klar, aber die Diskussion ueber CanBus und RS485 ist mir noch nicht verstaendlich.

Die ganze Diskussion war ueber die Gefahr, beim Ausfall der "Haupt Masse-Leitung", dass die Stroeme nach Alternativen Routen suchen.
Wie es z.Beispiel beim VE.Direkt Kabel des Victron MPPT passieren kann.
Vielleicht verstehe ich es nicht richtig (es wird nicht das erste Mal sein), aber ich sehe keine Grund warum beim Ausfall der "Haupt Masse-Leitung", solche Stroeme ueber CANbus oder RS485 entstehen koennten.

![Capture01](https://github.com/user-attachments/assets/5ac10824-baa8-4564-b655-33c349eabd68)

Hier ist meine CANbus Platine (eine von vielen)
Die Linke Seite der Platine arbeitet mit 3,3V vom ESP32. Damit sind RX/TX 3,3V Konform, es gibt nicht Mal die Sorge, dass die PINs der ESP32  mit 5V "ueberangestrengt" werden.
Auf der Linken Seite kommt ein High und ein Low Signal. Spannung (laut Google):

> Die Wellenform CAN-L wechselt von 2,5 V auf 1,5 V und die Wellenform CAN-H wechselt von 2,5 V auf 3,5 V.
Die niedrigen und hohen Spannungen und Übergänge zwischen ihnen weisen kein signifikantes Rauschen oder Verzerrungen auf.
Zwei vollständige CAN-Nachrichten, jeweils mit erhöhter Spannung am Ende.
Im Zeitraum zwischen den Nachrichten verbleiben CAN-L und CAN-H bei 2,5 V.
Man unterscheidet zwischen der CAN-Highspeed-Busankopplung (ISO 11898-2) und der CAN-Lowspeed-Busankopplung (ISO 11898-3).
CAN-Highspeed-Transceiver deuten innerhalb des Gleichtaktarbeitsbereichs, der üblicherweise zwischen 12 Volt und -12 Volt liegt

Ob wir High- oder Low-Speed CAN haben weiss ich nicht. Die Erklaerung aber gibt mir zu verstehen, dass CANbus unabhaengig von der Spannung des gesamten Systems (48V Batterie) funktioniert = Keine Gefahr (weder Stroeme, noch Spannung)




![SN65HVD230 CANbus Wiring different Boards](https://github.com/user-attachments/assets/396b6f8f-3064-4f59-80a2-03ada1f6817c)

**Reply by hoschiking** [reply]
Korrekt erkannt, die können auch nicht fließen, siehe meine Ausführungen weiter oben. Die Gefahr besteht dann nur darin, dass zu große Potenzialunterschiede zwischen den verbundenen Komponenten zu Schäden führen können. Also nichts der Strom ist dann das Problem, sondern die Spannung.

**Reply by SW-Niko** [reply]
Hallo @Manos1966,
jetzt mal ganz langsam. Ich erkläre dir mal die elektrotechnischen Hintergründe.

>CAN und RS485 haben keine Masse, sondern High und Low. 

Richtig. Das nennt sich symmetrische Übertragung. Du kannst damit Signale übertragen solange die Potentialdifferenz zwischen Sender und Empfänger eine bestimmte Spannung nicht überschreitet. Wenn ich deine Zeichnung von oben Verwende dann ist das diese Spannung, die dann am Empfänger gegen Masse auftritt.

![grafik](https://github.com/user-attachments/assets/d842dbf2-8248-46fd-973e-2f5dbe7b7be0)

Und wenn diese Spannung für den Empfangs IC zu hoch wird dann wird er beschädigt. RS485 ist spezifiziert für mindestens -7V bis +12V.
Ich bezweifle das er 48V verträgt.

@hoschiking Kannst du das bestätigen?

**Reply by SW-Niko** [reply]
@hoschiking Du bist immer schneller als ich. Aber gut das wir einer Meinung sind. 👍

**Reply by hoschiking** [reply]
Korrektur !!! 

das könnte gut sein, habe da was im Netz gefunden, was unsere Ausführungen bestätigt

![VCM](https://github.com/user-attachments/assets/13633c52-f1b3-4a47-b721-fb7105b954f9)

![VGPD](https://github.com/user-attachments/assets/d1871c3e-c81e-47cc-a59c-efea3b14e29e)

**Reply by SW-Niko** [reply]
>Ich denke, die -7V bis +12V beziehen sich auf die Spannungsdifferenz zwischen A+ und B-,

No.. Das findest du in Wikipedia

![grafik](https://github.com/user-attachments/assets/f75942b7-6250-46ed-a989-f008a768f012)

Aber in der Regel können die Treiber mehr. Der THDV2419 kann 42V.

![grafik](https://github.com/user-attachments/assets/cb03ab06-2594-4c71-acaa-5b87d223acb9)

**Reply by SW-Niko** [reply]
Schon wieder schneller ...

**Reply by SW-Niko** [reply]
Am besten wir fassen unsere Erkenntnisse zusammen und generieren ein PDF das die zusammenhänge erklärt.
Bist du dabei?

**Reply by hoschiking** [reply]
Danke, jetzt hätten wir also auch das geklärt.
Zusammenfassend kann man sagen:
Das Potenzial zwischen Sender/Empfänger darf -7 / +12V nicht übersteigen.
Das kann man sicherstellen durch  Potenzialausgleich zwischen Sender und Empfänger oder mindestens eines der beiden Seiten potentialfrei halten. Dann kann sich niemals ein Potenzialunterschied einstellen.

**Reply by hoschiking** [reply]
Jo bin gerne dabei, aber ich überlasse dir gerne den Vortritt und unterstütze im Hintegrund ;-)

**Reply by swingstate** [reply]
> ... oder mindestens eines der beiden Seiten potentialfrei halten.

Das erreiche ich mittels z.b. eines ADUM1201?

**Reply by SW-Niko** [reply]
@hoschiking 
> Jo bin gerne dabei, aber ich überlasse dir gerne den Vortritt und unterstütze im Hintergrund

Prima, Ich überarbeite mein Dokument und melde mich dann.

@swingstate 
> Das erreiche ich mittels z.b. eines ADUM1201? 

Für RS485? Mit einem ADUM1201 ja und nein.  Es kommt darauf an an welcher Stelle du trennst.
Der ADUM1201 ist für nicht symmetrisch Senden und Empfangen ausgelegt

![grafik](https://github.com/user-attachments/assets/c2215be4-f7fb-40c1-9b93-169301b6a588)

Um das richtig zu beantworten muss ich erst mal etwas nachforschen.

**Reply by hoschiking** [reply]
Das geht mittels ADUM, der muss nur zwischen UART und converter auf RS485, wenn einer verwendet wird. Die Spannungsversorgung und GND ist ab Ausgang des ADUM und für den rs485 vom Empfänger her zu verbinden. Ich hoffe, das ist halbwegs verständlich, sonst müsste ich es malen

**Reply by SW-Niko** [reply]
Hallo @hoschiking ,
ich habe einen Entwurf fertig. Im Anhang findest du eine OpenOffice Datei. Falls du ein anderes Format benötigst einfach melden.
Schreib deine Änderungen/Ergänzungen einfach in einer anderen Farbe rein und gib mir das Dokument zurück.
Danke. 
[ParalleleVerbindungen.odt](https://github.com/user-attachments/files/17874802/ParalleleVerbindungen.odt)

**Reply by hoschiking** [reply]
Hallo Niko,
sorry für die späte Antwort, ich war am WE unterwegs. Ich schaue es mir die Tage an und gebe dir Feedback. Das Format ist OK für mich

**Reply by hoschiking** [reply]
Hallo Niko,
Hab mir dein Dokument angeschaut, sehr gut geschrieben!! 2 Anmerkungen hatte ich, die sind gelb markiert.
Vielen Dank für deine tolle Arbeit.
[ParalleleVerbindungen.odt](https://github.com/user-attachments/files/17949943/ParalleleVerbindungen.odt)

**Reply by SW-Niko** [reply]
Hallo @hoschiking ,
vielen Dank für die Überprüfung. 👍

**Comment by Manos1966** [comment]
Ich sehe den Wald vor lauter Bäumen nicht @SW-Niko und @hoschiking 
Vielen Dank fuer die Erklaerungen, jetzt habe ich (einigermassen) verstanden 👍 

Es ist mir nicht klar was das Ziel ist. Ich dachte die Gefahr "Parallele Masseverbindungen".

Von eurer Erklaerungen verstehe ich es so, beim CANbus (und beim RS485 ?) gibt es keine Masse vom "Erzeuger" (die Batterie).
Der "Erzeuger" (die Batterie) hat ein System das High und Low Signale generiert. 
Dies ist ein "geschlossenes System", das bedeutet es hat keinen Kontakt zum restlichen 48V System der Batterie.
Also, im Fall wo ich das Haupt-Massen Kabel der Batterie trennen wuerde,  passiert nichts: Keine Stroeme/Spannungen werden alternative Routen suchen und bis auf meine OpenDTU durch die CANbus/RS485 Leitungen ankommen. Richtig?
Dann brauche ich keine galvanische Trennung fuer die RJ45.

**Comment by hoschiking** [comment]
Beispiel für eine Verbindung vom zum Beispiel ESP zum BMS. CAN analog dazu:
![20241108_205050](https://github.com/user-attachments/assets/c40dc315-86c4-40b2-b028-1859a8e0cfea)

**Reply by swingstate** [reply]
Du Schreibst CAN analog dazu. D.h. Sowohl für RS485 als auch einen CAN Transceiver würdest Du einen ADUM zwischen BMS  und die jeweiligen Transceiver bauen? Soll der ESP32 unten im Bild ein BMS darstellen?

**Reply by hoschiking** [reply]
Unten ist der ESP, oben das BMS. Ignoriere bitte die  Beschriftung unten links ;-)
Auf welcher Seite du den ADUM packst ist letztendlich egal, ich würde immer die ESP Seite bevorzugen, weil einfacher zu realisieren und kannst daran egal was anschließen.
Ob das bei CAN notwendig ist, kann ich nicht beurteilen, aber ich würde es tun, das Ding kostet bei Ali keine 2 Euro, darauf kommt es wirklich nicht an. Du kannst doch wie bei mir im Bild, jede UART (3) mit einem ADUM abschließen, was hinterher dran kommt ist dann letztendlich egal, somit weitestgehend universell. Dann muss man auch nicht jedes Mal die Verkabelung ändern wenn später Mal statt einem Pylontech was anderes angeschlossen wird. Im Sinne deiner Frage also ein ja, für beides

**Reply by swingstate** [reply]
Ich überlege dass auf einem PCB zu realisieren, werde das aber vorab einmal lose verkabelt testen - sowohl für CAN als auch RS485.

**Reply by hoschiking** [reply]
Eine gute Idee. Ich habe es nicht getestet, da ich beides nicht nutze.

**Comment by hoschiking** [comment]
Ich habe es etwas anders gelöst, aber nur weil ich mangels genug UART am ESP das BMS ausgelagert habe. Mein Versuchsaufbau mit 3x ADUM am ESP, daran 3x victron MPPT.
Einen weiteren ESP habe ich direkt am Speicher mit JK BMS, versorge den ESP mit Spannung direkt vom Speicher mit DC DC Wandler, hierdurch erschlage ich mehrere Fliegen mit einer Klappe. Keine galvanische Trennung notwendig, keine langen Datenkabel vom Speicher zum ESP. Der Speicher ist einfach mit 2 Leitungen an der Busbar angeschlossen, fertig. Auch eine Möglichkeit, die ich aufzeigen wollte.
![20241024_075903](https://github.com/user-attachments/assets/61425a4f-33b4-40d1-8949-ca3882e8f66e)

**Reply by swingstate** [reply]
Anfängerfrage - was tut der ESP am Speicher? Sehr schön verkabelt alles!

**Reply by hoschiking** [reply]
Der bindet das JK BMS mittels UART an, liest den Stromzähler per json aus und steuert den hoymiles , der aus dem Speicher ins Netz einspeist. Alles das ist direkt im Speicher eingebaut, aus dem nur 2 Kabel raus gehen. Keine galvanische Trennung notwendig, keine langen Kabel usw. 
Der ESP im Bild ist nur zum parsen von 3x MPPT und 2x Hoymiles von 2 Balkonkraftwerken. Die gesamte Steuerung läuft auf dem Raspi, die OpenDTU ist da völlig außen vor, bis auf die O-Einspeisung. Das ermöglicht sehr viele intelligente Lösungen, die so nicht möglich wären. Ich lade den Speicher per 4x MPPT, 2 direkt an Balkonkraftwerken, 2 AC gekoppelt von der etwas größeren Anlage auf dem Dach. Für die Steuerung der MPPT ist links im Bild der ESP 8266 zuständig, gesteuert über ioBroker.
Deswegen warte ich so sehnsüchtig, bis der @schlimmchen endlich eine Regelung des Ladestroms in Open DTUoB implementiert ;-) oder zumindest mqtt Topics um den MPPT am RX port aus und einschalten zu können

**Reply by Snusme83** [reply]
Servus zusammen, meine Adum1201 sind angekommen. Bräuchte Hilfe bezüglich Verbinden von Victron SmartSolar 100/20 - Adum1201 - ESP32-S3.
Aktuell benutze ich drei Laderegler ohne Adum mit 3 Adern --> GND, TX und RX. Muss jetzt VCC auch dazu?

Falls weiterhin nur 3 Adern:
VDD1, VOA, VIB, GND1 = Seite, an die der VE.direct vom Laderegler angeschlossen wird?

VDD2, VIA, VOB, GND2= Seite, die an den ESP32-S3 angeschlossen wird?

Falls dem so ist, muss:

RX vom Laderegler an VOA (Adum1201) und 
TX  vom Laderegler an VIB (Adum1201),oder?

**Reply by schlimmchen** [reply]
Steht in der [Doku](https://opendtu-onbattery.net/hardware/vedirect/#adum1201-safest-most-preferred), und Niko hat auch eine schöne Grafik gemacht. Die findest du in einem [Kommentar](https://github.com/hoylabs/OpenDTU-OnBattery/discussions/1394#discussioncomment-11272412) bis sie dann auch in der Doku landet.

**Reply by Snusme83** [reply]
@schlimmchen: Danke fürs Hinstupsen zur richtigen Übersicht! Hätte ich mal gewartet und auf deine Hilfe gewartet, hätte ich mir das Entlöten und Löten, Schrumpfschlauch,... gespart🤦🏻‍♂️

@SW-Niko: Danke für das tolle Bild und deine Bemühungen, was das Thema parallele Masseverbindungen/ Adum1201 angeht!
Mit diesem Bild, das anscheinend für Laderegler - Adum1201-ESP gilt hab ich es geschafft, den Adum1201 korrekt dazwischen zu löten:

![IMG_2979](https://github.com/user-attachments/assets/b250be8e-586b-49d0-99a5-0cf2b8aedfbd)

**Reply by Snusme83** [reply]
Ich habe nach diesem Bild gehandelt und das ist scheinbar für RS485- Verbindungen:
![IMG_2980](https://github.com/user-attachments/assets/aa69c0c8-b9a6-4023-a291-45c9a590ca32)

Frage: Wenn ich beim Pylontech bzw. Beim DC-/ DC- Wandler den Adum 1201 dazwischenlöte, handle ich nach diesem Bild, oder?

Für Laien wie mich wäre es hilfreich, das zu bebildern, wie oben beim Laderegler und ESP!

Wie gesagt, ganz tolle Arbeit. Das muss unbedingt ins Wiki!

**Reply by hoschiking** [reply]
Wie die Unterschrift im Bild besagt, ich das von dir genannten Bild nur ein Blockschaltbild.
Die Beschaltung ist immer identisch, egal was man miteinander verbinden möchte. Für symmetrische Signale wie RS 485 ist der ADUM tatsächlich nicht geeignet. Beim DC DC Wandler sowieso nicht, damit kannst du keine Leistung übertragen. Hierzu einfach die ganze Diskussion und Dokumentation durchlesen, darin solltest du alle nötigen Infos finden.

**Reply by SW-Niko** [reply]
Genau, hoschiking hat das schon richtig gesagt.
Für symmetrische Signale wie RS 485 ist der ADuM 1201 ungeeignet. 
Es geht schon, aber dann ist die galvanisch Isolierung hinfällig. 🤔 

Für symmetrische Signale kann der ADuM **1200** verwendet werden. Halb Duplex Betrieb.

Ich muss direkt mal nachschauen ... eine Voll Duplex CAN mit galvanischer Trennung habe ich schon mal gemacht.

**Reply by Snusme83** [reply]
Danke für die Rückmeldung. Aber mit diesen Begrifflichkeiten kommt ein Laie nicht klar.

Für mich ist jetzt klar, für die Verbindung von 3 Victron Ladereglern brauche ich drei Adum1201. Wie ich diese verbinden muss ist mir auch klar.

So, jetzt hab ich noch eine Verbindung zu den Pylontechs und eine Verbindung zum Breakoutboard welches von der Bubar mittels eines DC-/ DC Wandlers mit Strom versorgt wird!

Ist nicht so, dass ich ned viel lese hier, mir alles vorkauen lasse, oder mir ned selber Gedanken mache - aber gewisse Sachen sind halt auch trotz eurer Erklärung ned logisch!🤷🏻‍♂️
